Sources/Sentry/SentryFrameInAppLogic.m

```diff
@@ -0,0 +1,48 @@
+#import "SentryFrameInAppLogic.h"
+#import <Foundation/Foundation.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface
+SentryFrameInAppLogic ()
+
+@property (nonatomic, copy, readonly) NSArray<NSString *> *inAppIncludes;
+@property (nonatomic, copy, readonly) NSArray<NSString *> *inAppExcludes;
+
+@end
+
+@implementation SentryFrameInAppLogic
```

Comment:
Why can't this be a method somewhere instead of being its own type?
It's literally 2 `for` loops. IMHO it looks overengineered.

https://github.com/getsentry/sentry-dotnet/blob/df87643ddc94afd9312a15dd34df3bc5b93f0136/src/Sentry/Extensibility/SentryStackTraceFactory.cs#L222-L223

https://github.com/getsentry/sentry-dart/blob/56412f9a351f21842d6242b97d0a47576ca35c66/dart/lib/src/sentry_stack_trace_factory.dart#L153-L173

https://github.com/getsentry/sentry-java/blob/8324e225c2d9217ccca39da3ad1f5441856677da/sentry/src/main/java/io/sentry/SentryStackTraceFactory.java#L76-L95

Reply:
The problem is that I need the logic in two different places, `SentryCrashReportConverter` and `SentryCrashStackEntryMapper`. In those two places, I don't have the `SentryOptions`, which I need for the `inAppIncludes` and `inAppExcludes`. So either I would pass the options where I now pass the `SentryFrameInAppLogic` or I would pass `inAppIncludes` and `inAppExcludes`, which is definitely not nice, or I put them inside `SentryFrameInAppLogic`. I went for the last option. Would you prefer passing the options around instead and making `SentryFrameInAppLogic` a simple method?

Reply:
That's cool I didn't look through in detail enough to spot the multiple usages of it.

Reply:
Thanks for raising such a valid point anyway. I'm happy for such feedback.